homeassistant/components/climate/generic_thermostat.py

```diff
@@ -59,140 +65,197 @@
+                            float(old_state.attributes[ATTR_TEMPERATURE])
+                    if self._target_temp_high is None:
+                        self._target_temp_high = \
+                            float(old_state.attributes[ATTR_TEMPERATURE])
+
+            # Restore _is_away
             if old_state.attributes.get(ATTR_AWAY_MODE) is not None:
                 self._is_away = str(
                     old_state.attributes[ATTR_AWAY_MODE]) == STATE_ON
-            if (self._initial_operation_mode is None and
-                    old_state.attributes[ATTR_OPERATION_MODE] is not None):
+
+            # Restore _current_operation from previous value
+            if (self._current_operation is None and
+                    old_state.attributes.get(ATTR_OPERATION_MODE)
+                    is not None):
                 self._current_operation = \
                     old_state.attributes[ATTR_OPERATION_MODE]
-                self._enabled = self._current_operation != STATE_OFF
 
-        else:
-            # No previous state, try and restore defaults
-            if self._target_temp is None:
-                if self.ac_mode:
-                    self._target_temp = self.max_temp
-                else:
-                    self._target_temp = self.min_temp
-            _LOGGER.warning("No previously saved temperature, setting to %s",
-                            self._target_temp)
+        # Fill value with defaults if not found in old_state
+        if self._target_temp_high is None:
+            self._target_temp_high = DEFAULT_TARGET_TEMP_HIGH
+        if self._target_temp_low is None:
+            self._target_temp_low = DEFAULT_TARGET_TEMP_LOW
+        if self._saved_target_temp_high is None:
+            self._saved_target_temp_high = DEFAULT_TARGET_TEMP_HIGH
+        if self._saved_target_temp_low is None:
+            self._saved_target_temp_low = DEFAULT_TARGET_TEMP_LOW
+        if self._current_operation not in self.operation_list:
+            self.set_operation_mode(STATE_OFF)
+
+        if STATE_AUTO in self.operation_list and \
+                (self._target_temp_low + self._hot_tolerance >=
+                 self._target_temp_high - self._cold_tolerance):
+            _LOGGER.error("Low and high too close."
+                          "Unpredictable results with %s - %s",
+                          self._target_temp_low, self.target_temperature_high)
 
     @property
     def state(self):
         """Return the current state."""
-        if self._is_device_active:
-            return self.current_operation
-        if self._enabled:
+        if self.current_operation == STATE_OFF:
+            return STATE_OFF
+        if self.is_on:
+            if self.current_operation == STATE_AUTO:
+                if self._is_heating and self._is_cooling:
+                    # Emergency shut off
+                    _LOGGER.error("Both heat and cool on, turning off")
+                    self.set_operation_mode(STATE_OFF)
```

Comment:
Properties should never do I/O

Reply:
I can't find a way to catch if someone manually switches on both switches from outside the generic thermostat. This set_operation_mode was used to have a safety switch off. 
I'll keep looking but if you have a suggestion I'll happily take it